suite-common/wallet-utils/src/transactionUtils.ts

```diff
@@ -42,20 +43,16 @@ const generateTransactionDateKey = (d: Date) =>
 
 /** Parse Date object from a string in YYYY-MM-DD format */
 export const parseTransactionDateKey = (key: string) => {
-    const parts = key.split('-');
-    const d = new Date(Number(parts[0]), Number(parts[1]) - 1, Number(parts[2]));
-    return d;
+    const [year, month, day] = key.split('-');
+    return new Date(Number(year), Number(month) - 1, Number(day));
 };
 
-/* Convert date to string in YYYY-MM format */
-const generateTransactionMonthKey = (d: Date) => `${d.getFullYear()}-${d.getMonth() + 1}`;
+export type DatetimeKey = string & { __type: 'DatetimeKey' };
+const generateTransactionDatetimeKey = (d: Date): DatetimeKey =>
+    // Adding days because of time zones of UTC
+    addDays(startOfMonth(d), 1).toUTCString() as DatetimeKey;
 
-/** Parse Date object from a string in YYYY-MM format */
-export const parseTransactionMonthKey = (key: string) => {
-    const parts = key.split('-');
-    const d = new Date(Number(parts[0]), Number(parts[1]) - 1);
-    return d;
-};
+export const parseTransactionDateTimeKey = (key: DatetimeKey): Date => new Date(key);
```

Comment:
Why did you renamed it to "datetime" everywheme? It doesn't make sense now, it's still just `MonthKey` and you do not care if it uses datetime under-the-hood.